CoreWebServer/src/au/gov/asd/tac/constellation/webserver/restapi/RestServiceUtilities.java

```diff
@@ -170,36 +171,22 @@ public static String activeGraphId() {
      *
      * @return The new graph id.
      */
-    public static String waitForGraphChange(final String existingId) {
-        // Now we wait for the new graph to become active.
-        //
-        int waits = 0;
-        while (true) {
-            // Wait a bit to give the new graph time to become active.
-            //
-            try {
-                Thread.sleep(1000);
-            } catch (final InterruptedException ex) {
-                Thread.currentThread().interrupt();
-                throw new RestServiceException(ex);
-            }
-
-            final Graph newGraph = GraphManager.getDefault().getActiveGraph();
-            final String newId = newGraph != null ? newGraph.getId() : null;
-            if ((existingId == null && newId != null) || (existingId != null && newId != null && !existingId.equals(newId))) {
-                // - there was no existing graph, and the new graph is active, or
-                // - there was an existing graph, and the active graph is not the existing graph.
-                // - we assume the user hasn't interfered by manually switching to another graph at the same time.
-                //
-                return newId;
-            }
-
-            // We only have so much patience.
-            //
-            if (++waits > 10) {
-                throw new RestServiceException("The new graph has taken too long to become active");
+    public static CompletableFuture<String> waitForGraphChange(final String existingId) {
+
+        return CompletableFuture.supplyAsync(() -> {
+            while (true) {
+                if (GraphManager.getDefault().getActiveGraph() != null) {
+                    final Graph newGraph = GraphManager.getDefault().getActiveGraph();
+                    final String newId = newGraph != null ? newGraph.getId() : null;
```

Comment:
the if statement this sits in already checks that the active graph (which is what `newGraph` is set to) is not null so this check shouldn't be necessary (especially if combined with the above comment)

Reply:
This check can still be removed (as the check is already done to get into the if block)